pkg/operator/controller/garden/garden/reconciler_delete.go

```diff
@@ -84,6 +84,10 @@ func (r *Reconciler) delete(
 			Fn:   component.OpDestroyAndWait(c.blackboxExporter).Destroy,
 		})
 
+		destroyGardenerDiscoveryServer = g.Add(flow.Task{
```

Comment:
In the reconciliation, you wait for GAPI readiness. For symmetry reasons, shouldn't we wait for GAPI deletion here?

Reply:
> In the reconciliation, you wait for GAPI readiness.

Yes, because the discovery server retrieves shoots and projects in order to perform certain validations.

> For symmetry reasons, shouldn't we wait for GAPI deletion here?

Since the discovery server uses the Shoot & Project APIs it would make sense for the GAPI to wait the Discovery Server tear down so that exits gracefully. But that seemed a little awkward and it should not matter much since the whole garden is being deleted anyway. So I would vote for one of the following:
- leave it as it is
- wait for discovery server deletion before deleting the GAPI

WDYT?

Reply:
OK, then let's just leave it as it is.